WordPress/Classes/Utility/Media/MediaImageExporter.swift

```diff
@@ -0,0 +1,257 @@
+import Foundation
+import MobileCoreServices
+
+/// MediaLibrary export handling of UIImages.
+///
+class MediaImageExporter: MediaExporter {
+
+    /// Default filename used when writing media images locally, which may be appended with "-1" or "-thumbnail".
+    ///
+    let defaultImageFilename = "image"
+
+    var maximumImageSize: CGFloat?
+    var stripsGeoLocationIfNeeded = false
+    var mediaDirectoryType: MediaLibrary.MediaDirectoryType = .uploads
+
+    public enum ExportError: MediaExportError {
+        case imageJPEGDataRepresentationFailed
+        case imageSourceCreationWithDataFailed
+        case imageSourceCreationWithURLFailed
+        case imageSourceIsAnUnknownType
+        case imageSourceExpectedJPEGImageType
+        case imageSourceDestinationWithURLFailed
+        case imageSourceThumbnailGenerationFailed
+        case imageSourceDestinationWriteFailed
+        var description: String {
+            switch self {
+            case .imageJPEGDataRepresentationFailed,
+                 .imageSourceCreationWithDataFailed,
+                 .imageSourceCreationWithURLFailed,
+                 .imageSourceIsAnUnknownType,
+                 .imageSourceExpectedJPEGImageType,
+                 .imageSourceDestinationWithURLFailed,
+                 .imageSourceThumbnailGenerationFailed,
+                 .imageSourceDestinationWriteFailed:
+                return NSLocalizedString("The image could not be added to the Media Library.", comment: "Message shown when an image failed to load while trying to add it to the Media library.")
+            }
+        }
+        func toNSError() -> NSError {
+            return NSError(domain: _domain, code: _code, userInfo: [NSLocalizedDescriptionKey: String(describing: self)])
+        }
+    }
+
+    /// Exports and writes a UIImage to a local Media URL.
+    ///
+    /// A JPEG or PNG is expected, but not necessarily required. Exporting will fail if a JPEG cannot
+    /// be represented from the UIImage, such as trying to export a GIF.
+    ///
+    /// - parameter fileName: Filename if it's known.
+    /// - parameter onCompletion: Called on successful export, with the local file URL of the exported UIImage.
+    /// - parameter onError: Called if an error was encountered during creation.
+    ///
+    func exportImage(_ image: UIImage, fileName: String?, onCompletion: @escaping (MediaImageExport) -> (), onError: @escaping (MediaExportError) -> ()) {
+        do {
+            guard let data = UIImageJPEGRepresentation(image, 1.0) else {
```

Comment:
Are we sure we want maximum quality here? 
and shouldn't this be a parameter of the method? It could be added with a default value of 0.9